scripts/core/editor3/actions/spellchecker.jsx

```diff
@@ -21,14 +16,61 @@ export function replaceWord(word) {
 
 /**
  * @ngdoc method
- * @name Spellchecker Actions#showContextMenu
- * @param {Object} contextMenuData
+ * @name replaceWord
+ * @param {String} word
+ * @return {String} action
+ * @description Creates the add word action
+ */
+export function addWord(word) {
+    const spellcheck = ng.get('spellcheck');
+
+    spellcheck.addWord(word, false);
+
+    return replaceWord(word);
+}
+
+/**
+ * @ngdoc method
+ * @name replaceWord
+ * @param {String} word
+ * @return {String} action
+ * @description Creates the ignore word action
+ */
+export function ignoreWord(word) {
+    const spellcheck = ng.get('spellcheck');
+
+    spellcheck.addWord(word, true);
+
+    return replaceWord(word);
```

Comment:
Maybe, but keeping a huge dictionary in the state could potentially raise a performance issue.